langlib/lang.array/src/main/java/org/ballerinalang/langlib/array/Sort.java

```diff
@@ -42,12 +48,21 @@
 public class Sort {
 
     public static BArray sort(BArray arr, Object direction, Object func) {
-        checkIsArrayOnlyOperation(TypeUtils.getImpliedType(arr.getType()), "sort()");
+        BArray sortedArray;
         BFunctionPointer<Object, Object> function = (BFunctionPointer<Object, Object>) func;
-
+        Set<Type> typeList = new HashSet<>();
+        if (arr.getType().getTag() == TypeTags.TUPLE_TAG) {
+            if (!isOrderedType(arr.getType()) && function == null) {
+                throw ErrorCreator.createError(getModulePrefixedReason(ARRAY_LANG_LIB, INVALID_TYPE_TO_SORT),
+                        StringUtils.fromString("Valid key function required"));
```

Comment:
I would elaborate on this a bit to include why it is needed (i.e., members don't belong to a single ordered type).

Reply:
Fixed with c5daaea654d92b306868ce09652d183a764da99f